OpenRA.Mods.Common/Widgets/ObserverArmyIconsWidget .cs

```diff
@@ -0,0 +1,220 @@
+
+			Init();
+		}
+
+		protected ObserverArmyIconsWidget(ObserverArmyIconsWidget other)
+			: base(other)
+		{
+			GetPlayer = other.GetPlayer;
+			world = other.world;
+			worldRenderer = other.worldRenderer;
+
+			IconWidth = other.IconWidth;
+			IconHeight = other.IconHeight;
+			IconSpacing = other.IconSpacing;
+			iconSize = new float2(IconWidth, IconHeight);
+
+			MinWidth = other.MinWidth;
+
+			TooltipUnit = other.TooltipUnit;
+			GetTooltipUnit = () => TooltipUnit;
+
+			TooltipTemplate = other.TooltipTemplate;
+			TooltipContainer = other.TooltipContainer;
+
+			tooltipContainer = Exts.Lazy(() =>
+				Ui.Root.Get<TooltipContainerWidget>(TooltipContainer));
+
+			Init();
+		}
+
+		void Init()
+		{
+			var buildable = world.Map.Rules.Actors.Values
+				.Where(x =>
+					x.Name[0] != '^' &&
+					x.HasTraitInfo<BuildableInfo>())
+				.SelectMany(x => x.TraitInfo<BuildableInfo>().Queue, (ai, q) => new Tuple<string, ActorInfo, BuildableInfo>(q, ai, ai.TraitInfo<BuildableInfo>()));
+
+			var pqueue = world.Map.Rules.Actors.Values
+				.SelectMany(a => a.TraitInfos<ProductionQueueInfo>());
+
+			unitLookup = buildable.Join(pqueue, tuple => tuple.Item1, pq => pq.Type, (tuple, info) => new { tuple.Item2.Name, Ai = tuple.Item2, tuple.Item3.BuildPaletteOrder, info.DisplayOrder })
+				.Distinct()
+				.ToDictionary(t => t.Name, o => new Tuple<ActorInfo, int, int>(o.Ai, o.DisplayOrder, o.BuildPaletteOrder));
+		}
+
+		public override void Draw()
+		{
+			armyUnits.Clear();
+			armyIconsBounds.Clear();
+
+			var player = GetPlayer();
+			if (player == null)
+				return;
+
+			var playerStatistics = player.PlayerActor.TraitOrDefault<PlayerStatistics>();
+
+			var items = playerStatistics.Units
+				.Where(i => i.Value.Count > 0)
+				.OrderBy(i => unitLookup[i.Key].Item2)
+				.ThenBy(i => unitLookup[i.Key].Item3)
+				.ToList();
+
+			Bounds.Width = items.Count * (IconWidth + IconSpacing);
+
+			var queueCol = 0;
+			foreach (var unit in items)
+			{
+				var name = unit.Key;
+				var actorInfo = unitLookup[name].Item1;
+
+				var rsi = actorInfo.TraitInfo<RenderSpritesInfo>();
+				var icon = new Animation(world, rsi.GetImage(actorInfo, world.Map.Rules.Sequences, player.Faction.Name));
+				var bi = actorInfo.TraitInfo<BuildableInfo>();
+				icon.Play(bi.Icon);
+				var topLeftOffset = new float2(queueCol * (IconWidth + IconSpacing), 0);
+
+				var iconTopLeft = RenderOrigin + topLeftOffset;
+				var centerPosition = iconTopLeft;
+
+				WidgetUtils.DrawSHPCentered(icon.Image, centerPosition + 0.5f * iconSize, worldRenderer.Palette(bi.IconPalette), 0.5f);
+
+				armyUnits.Add(actorInfo);
+				armyIconsBounds.Add(new Rectangle((int)iconTopLeft.X, (int)iconTopLeft.Y, (int)iconSize.X, (int)iconSize.Y));
+
+				var bold = Game.Renderer.Fonts["Regular"];
+				var text = unit.Value.Count.ToString();
+				bold.DrawTextWithContrast(text, centerPosition + new float2(16, 0) - new float2(bold.Measure(text).X / 2, 0),
```

Comment:
I find the visibility of these to be really poor, but i don't have any ideas on how to improve it (other than going to the lowres icons that have a cleaner image, which you've already ruled out)

Reply:
Added some more offset made it more visible.

![image](https://user-images.githubusercontent.com/19173272/61745054-fcd59200-ad98-11e9-86fa-ff9f806df53c.png)

Reply:
And with offset 3
![image](https://user-images.githubusercontent.com/19173272/61748388-5ee5c580-ada0-11e9-9825-b57e1a81c7c0.png)

Reply:
TinyBold with offset 2
![image](https://user-images.githubusercontent.com/19173272/61748551-c7cd3d80-ada0-11e9-97dc-7d9ed8dc1161.png)

Reply:
Or manual resize icons
![icons](https://user-images.githubusercontent.com/19173272/61749195-758d1c00-ada2-11e9-8820-e092b65a19ff.png)

Reply:
You should be able to get a similar effect without custom assets by toggling `Sprite.Sheet.GetTexture().ScaleFilter = TextureScaleFilter.Linear;`, but you'll need to do some special bookkeeping at load time to make sure the icons end up on their own sheet that isn't shared with other things.

Reply:
The possibility to use manual resize icons sounds good IMO - modders could then supplement manually downscaled textless cameos for this.

Reply:
Something did not work with the TextureScaleFilter.Linear;, 
![image](https://user-images.githubusercontent.com/19173272/61822802-c19b9780-ae5a-11e9-951d-ad901a149935.png)

Reply:
Ah right of course, this will only work for RGBA sprites. For indexed sprites it does bilinear interpolation on the palette indices which just scrambles everything.

Reply:
Think this is the best we can get while keeping the Icon size
![image](https://user-images.githubusercontent.com/19173272/61824755-16411180-ae5f-11e9-88ca-4f41e837f171.png)

Reply:
![image](https://user-images.githubusercontent.com/19173272/61827948-f103d180-ae65-11e9-9745-fc3e0979aa5e.png)